### DISCUSSION TITLE
Scrapli Proxy Connections

### DISCUSSION THREAD

**Discussion by tybruno** [carlmontanari/scrapli#161] [Ideas]
Hello,

I need to handle proxy jump host connections to connect to protected devices. It seems that Scrapli can handle proxy connections by defining the proxy connection in the ssh config file when using asyncssh. 

Is there another way to use a proxy connection with Scrapli? At run time I need the user to be able to define the proxy connection and creds at run time. This would be challenging having to use config files.

**Comment by carlmontanari** [comment]
For system transport I think you can probably extend the "open cmd" to include any proxy directives at runtime. (I think, not 100% sure)

For asyncssh I think there is no way (in scrapli) to get at the connection prior to opening to augment/modify.

But... You could probably just prompt the user and get whatever info, then write it out to a temp ssh config file and clean it up when you're done. That seems like it may be a viable option.... but maybe not 😁
